src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
-                            x => x.GracefulStop(settings.FlushWait, EndpointWriter.FlushAndStop.Instance))).ContinueWith(
-                                result =>
+                    var shutdownStatus = Task.WhenAll(_endpoints.AllEndpoints.Select(
+                        x => x.GracefulStop(_settings.FlushWait, EndpointWriter.FlushAndStop.Instance))).ContinueWith(
+                            result =>
+                            {
+                                if (result.IsFaulted || result.IsCanceled)
                                 {
-                                    if (result.IsFaulted || result.IsCanceled)
-                                    {
-                                        if (result.Exception != null)
-                                            result.Exception.Handle(e => true);
-                                        return false;
-                                    }
-                                    return result.Result.All(x => x);
-                                }, TaskContinuationOptions.ExecuteSynchronously);
-
-                    shutdownStatus.ContinueWith(tr => Task.WhenAll(_transportMapping.Values.Select(x => x.Shutdown())).ContinueWith(
-                              result =>
+                                    if (result.Exception != null)
+                                        result.Exception.Handle(e => true);
+                                    return false;
+                                }
+                                return result.Result.All(x => x);
+                            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+
+                shutdownStatus.ContinueWith(tr => Task.WhenAll(_transportMapping.Values.Select(x => x.Shutdown())).ContinueWith(
+                          result =>
+                          {
+                              if (result.IsFaulted || result.IsCanceled)
                               {
-                                  if (result.IsFaulted || result.IsCanceled)
-                                  {
-                                      if (result.Exception != null)
-                                          result.Exception.Handle(e => true);
-                                      return false;
-                                  }
-                                  return result.Result.All(x => x) && tr.Result;
-                              }, TaskContinuationOptions.ExecuteSynchronously)).Unwrap().PipeTo(sender);
+                                  if (result.Exception != null)
+                                      result.Exception.Handle(e => true);
+                                  return false;
+                              }
+                              return result.Result.All(x => x) && tr.Result;
+                          }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)).Unwrap().PipeTo(sender);
```

Comment:
Unwrap the inner task for shutting down all of the transports and pipe the results back to the sender.